packages/expo-av/android/src/main/java/expo/modules/av/AVManager.java

```diff
@@ -669,6 +676,135 @@ public void prepareAudioRecorder(final ReadableArguments options, final Promise
     promise.resolve(map);
   }
 
+  private AudioDeviceInfo getDeviceInfoFromUid(String uid) {
+    AudioDeviceInfo deviceInfo = null;
+    if (android.os.Build.VERSION.SDK_INT < android.os.Build.VERSION_CODES.M) {
+      return deviceInfo;
+    }
+
+    int id = Integer.valueOf(uid);
+    AudioDeviceInfo[] audioDevices = mAudioManager.getDevices(AudioManager.GET_DEVICES_INPUTS);
+    for (int i = 0; i < audioDevices.length; i++) {
+      AudioDeviceInfo device = audioDevices[i];
+      int deviceId = device.getId();
+      if (deviceId == id) {
+        deviceInfo = device;
+      }
+    }
+    return deviceInfo;
```

Comment:
We could do it a bit simpler:

```suggestion
    for (AudioDeviceInfo device : audioDevices) {
      int deviceId = device.getId();
      if (deviceId == id) {
        return device;
      }
    }
```